specification/terraform/Microsoft.AzureTerraform.Management/models.tsp

```diff
@@ -92,45 +92,51 @@ model ExportResourceGroup extends BaseExportModel {
   type: "ExportResourceGroup";
 }
 
+// Not using Azure.ResourceManager.ArmOperationStatus due to bug in the API response where id is a long URL instead of uuid
+// Will be fixed at next API version
 @doc("The status of the LRO operation.")
-model OperationStatus {
-  @doc("The operation status resource id.")
-  id?: string;
+model TerraformArmOperationStatus<
+  Properties extends {} = never,
+  StatusValues extends TypeSpec.Reflection.Union = ResourceProvisioningState
+> {
+  /** RP-specific properties for the operationStatus resource, only appears when operation ended with Succeeded status */
+  @visibility("read")
+  properties?: Properties;
+
+  /** The operation status */
+  @Azure.Core.lroStatus
+  status: StatusValues;
 
-  @doc("The fully qualified resource id of the resource for which the operation was performed.")
+  /** The unique identifier for the operationStatus resource */
+  @key
+  @path
+  @segment("operationStatuses")
   @visibility("read")
-  resourceId?: string;
+  id: string;
 
-  @doc("The operation name.")
+  /** The name of the  operationStatus resource */
+  @visibility("read")
   name?: string;
 
-  @doc("The start time of the operation.")
+  /** Operation start time */
   @visibility("read")
   startTime?: utcDateTime;
 
-  @doc("The end time of the operation.")
+  /** Operation complete time */
   @visibility("read")
   endTime?: utcDateTime;
 
-  @doc("The status of the operation.")
-  status?: string;
-
-  @doc("The progress percentage of the operation, ranges from 0 to 100")
+  /** The progress made toward completing the operation */
+  @visibility("read")
   percentComplete?: float64;
 
-  @doc("The Terraform export result")
-  properties?: ExportResult;
-
-  ...ErrorResponse;
+  /** Errors that occurred if the operation ended with Canceled or Failed status */
+  @visibility("read")
+  error?: ErrorDetail;
 }
 
-model InProgressOperationStatus
-  is ArmAcceptedResponse<
-    "InProgress operation status",
-    ArmCombinedLroHeaders & Azure.Core.Foundations.RetryAfterHeader
-  > {
-  ...OperationStatus;
-}
+@doc("The status of the LRO operation.")
+model OperationStatus is TerraformArmOperationStatus<ExportResult>;
```

Comment:
and this would make more sense if this name is `TerraformOperationStatus`

Reply:
Renamed to TerraformOperationStatus in 99daad74c1